src/main/java/seedu/address/model/export/JsonExportPath.java

```diff
@@ -0,0 +1,147 @@
+//@@author LeowWB
+
+package seedu.address.model.export;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.AppUtil.checkArgument;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.List;
+
+import seedu.address.model.flashcard.FlashCard;
+
+/**
+ * Represents the full path to a JSON export file, including parent directories.
+ * Guarantees: immutable; is valid as declared in {@link #isValid(String)}
+ */
+public class JsonExportPath extends ExportPath {
+
+    public static final String MESSAGE_CONSTRAINTS =
+            "JSON export file path may only consist of alphanumeric characters, spaces, and the following characters:\n"
+            + "~\\/-_!:[]()\n"
+            + "It must also end with \".json\".";
+
+    /*
+     * The following characters are allowed (in addition to alphanumeric):
+     * ~\/-_!:[]()
+     * Space is allowed.
+     * Required to end with the String: ".json"
+     */
+    public static final String VALIDATION_REGEX = "[.~\\w\\-!:\\[\\]()/\\\\ ]+\\.[Jj][Ss][Oo][Nn]";
+
+    private final DirectoryPath directoryPath;
+    private final JsonExportFilePath jsonExportFilePath;
+
+    /**
+     * Constructs a {@code JsonExportPath}.
+     *
+     * @param jsonExportPath A valid JSON export path.
+     */
+    public JsonExportPath(String jsonExportPath) {
+        requireNonNull(jsonExportPath);
+        checkArgument(isValid(jsonExportPath), MESSAGE_CONSTRAINTS);
+        this.directoryPath = extractDirectoryPath(jsonExportPath);
```

Comment:
Avoid unnecessary use of `this` keyword.

Reply:
I agree that it's redundant, but I'll choose to keep it there so that the code retains a parallel structure to the next line (line 47).